filesystem/iso9660.ksy

```diff
@@ -7,269 +7,959 @@ meta:
+      directory_record:
+        doc-ref: ecma-119 9.1
+        seq:
+          - id: len_dr
+            doc-ref: ecma-119 9.1.1
+            doc: |
+              If len_dr == 0 we do not process the body
+              If len_dr >= 1 we include field len_dr in the size
+            type: u1
+          - id: body
+            type: body
+            if: len_dr > 0x0
+        types:
+          body:
+            seq:
+              - id: ext_attr_rec_len
+                doc-ref: ecma-119 9.1.2
+                type: u1
+              - id: location_of_extent
+                doc-ref: ecma-119 9.1.3
+                type: u4bi
+              - id: data_len
+                doc-ref: ecma-119 9.1.4
+                type: u4bi
+              - id: rec_date_time
+                doc-ref: ecma-119 9.1.5
+                type: datetime_short
+              - id: file_flags_multi_extent
+                doc-ref: ecma-119 9.1.6 b7
+                type: b1
+              - id: file_flags_reserved
+                doc-ref: ecma-119 9.1.6 b5+b6
+                type: b2
+              - id: file_flags_protection
+                doc-ref: ecma-119 9.1.6 b4
+                type: b1
+              - id: file_flags_record
+                doc-ref: ecma-119 9.1.6 b3
+                type: b1
+              - id: file_flags_associated_file
+                doc-ref: ecma-119 9.1.6 b2
+                type: b1
+              - id: file_flags_directory
+                doc-ref: ecma-119 9.1.6 b1
+                type: b1
+              - id: file_flags_existence
+                doc-ref: ecma-119 9.1.6 b0
+                type: b1
+              - id: file_unit_size
+                doc-ref: ecma-119 9.1.7
+                type: u1
+              - id: interleave_gap_size
+                doc-ref: ecma-119 9.1.8
+                type: u1
+              - id: vol_seq_num
+                doc-ref: ecma-119 9.1.9
+                type: u2bi
+              - id: len_fi
+                doc-ref: ecma-119 9.1.10
+                type: u1
+              - id: file_id_file
+                doc-ref: ecma-119 9.1.11
+                type: str
+                size: len_fi
+                if: not file_flags_directory
+              - id: file_id_dir
+                doc-ref: ecma-119 9.1.11
+                size: len_fi
+                type: str
+                if: file_flags_directory
+              - id: padding_field
+                doc-ref: ecma-119 9.1.12
+                doc: |
+                  Padding field is added when len_fi contains an even number
+                contents: [ 0x0 ]
+                if: len_fi % 2 == 0
+              - id: system_use
+                doc-ref: ecma-119 9.1.13
+                doc: |
+                  The size of the system_use is defined as:
+                  ( len_dr - size of directory_record = 33 bytes ) - len_fi
+                type: susp
+                size: ( _parent.len_dr - 33 ) - len_fi
```

Comment:
This is actually not correct: `susp` is whatever it is from up to that point, to the end of the directory record. If there is a padding byte, then it should be decreased by 1.

By specifying the size it becomes really easy to avoid this problem:

```
                size-eos: true
```